### PR TITLE
OCP4: Add rule to check that the default policy is to reject images

### PR DESCRIPTION
**Pull request by JAORMX**
Container runtimes have the capability of specifying what images to
accept and what policy to use for each registry. The intent is to ensure
that integrity verification is enabled for relevant repos.

We, thus, create a rule to ensure that the default policy is to reject,
and that administrators explicitly enable each repo they're actually
interested in using. This addresses several NIST controls which are
linked on the rule.

The remediation introduces a policy that ensures that Red Hat registries
are signature-verified, but that is lenient towards the images we use
for testing (e.g. coming from OpenShift CI). We cannot create an
automated remediation here as this configuration is very
deployment-specific and we cannot guess what registries are customers
using, the signature type and policy they'd like to enforce. However, at
least we can get them thinking.

Signed-off-by: Juan Antonio Osorio Robles <jaosorior@redhat.com>

### CODE REVIEW

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 13760659e76b268d7ad66dfca8c9da84258b49b7

**Review by jhrozek — APPROVED** [review]
/lgtm

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now e5eb8b680c0eb20ce10c9f18428b3cff18f316d3

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 85d511fe4a6d693e99c757dc7de7e1d1375ef556

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now a30af157f2fc033eef8f0114b682ccf0a18e69a3

**Review by Vincent056 — APPROVED** [review]

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 4afb1e205b45e58035be4eef4e64f8de54af9460

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 50420820146e999220e7b6f4b1340fca28d316ea

File: tests/ssg_test_suite/xml_operations.py [hunk 1]
```diff
@@ -80,21 +80,39 @@ def datastream_root(ds_location, save_location=None):
...
-    remove_machine_only_from_element(root, "xccdf-1.2:Group")
-
-
-def remove_machine_only_from_element(root, element_spec):
+def remove_platforms_from_element(root, element_spec, platforms):
     query = BENCHMARK_QUERY + "//{0}".format(element_spec)
     elements = root.findall(query, PREFIX_TO_NS)
     for el in elements:
         platforms = el.findall("./xccdf-1.2:platform", PREFIX_TO_NS)
         for p in platforms:
```
Comment (ggbecker) [inline tests/ssg_test_suite/xml_operations.py:88]:
```suggestion
        platforms_xml = el.findall("./xccdf-1.2:platform", PREFIX_TO_NS)
        for p in platforms_xml:
```

it's conflicting with the new `platforms` you have just created.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now b9ba262181fb80af8deb78d9963de70e14300af7

**Pull request merged**